content/en/platform/corda/5.0/developing-applications/api/ledger/utxo-ledger/vault-queries.md

```diff
@@ -0,0 +1,561 @@
+    override fun verify(transaction: UtxoLedgerTransaction) {}
+}
+
+@BelongsToContract(TestContract::class)
+class TestState(
+    val testField: String,
+    private val participants: List<PublicKey>
+) : ContractState {
+    override fun getParticipants(): List<PublicKey> = participants
+}
+```
+
+{{% /tab %}}
+{{% tab name="Java" %}}
+
+```java
+package com.r3.corda.demo.contract;
+
+class TestContract implements Contract {
+    @Override
+    public void verify(@NotNull UtxoLedgerTransaction transaction) {}
+}
+
+@BelongsToContract(TestContract.class)
+public class TestState implements ContractState {
+
+    private final List<PublicKey> participants;
+    private final String testField;
+
+    public TestState(@NotNull String testField, @NotNull List<PublicKey> participants) {
+        this.testField = testField;
+        this.participants = participants;
+    }
+
+    @NotNull
+    @Override
+    public List<PublicKey> getParticipants() {
+        return new LinkedList<>(this.participants);
+    }
+
+    @NotNull
+    public String getTestField() {
+        return testField;
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+This contract has no verification logic and should only be used for testing purposes. The state itself has a `testField` property defined for JSON representation and constructing queries.
+
+To represent a state as a JSON string, use `ContractStateVaultJsonFactory` as follows:
+
+{{< tabs name="tabs-2" >}}
+{{% tab name="Kotlin" %}}
+
+```kotlin
+class TestStateJsonFactory : ContractStateVaultJsonFactory<TestState> {
+    override fun getStateType(): Class<TestState> = TestState::class.java
+
+    override fun create(state: TestState, jsonMarshallingService: JsonMarshallingService): String {
+        return jsonMarshallingService.format(state)
+    }
+}
+```
+{{% /tab %}}
+{{% tab name="Java" %}}
+
+```java
+class TestUtxoStateJsonFactory implements ContractStateVaultJsonFactory<TestUtxoState> {
+    @NotNull
+    @Override
+    public Class<TestUtxoState> getStateType() {
+        return TestUtxoState.class;
+    }
+
+    @Override
+    @NotNull
+    public String create(@NotNull TestUtxoState state, @NotNull JsonMarshallingService jsonMarshallingService) {
+        return jsonMarshallingService.format(state);
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+After the output state has been finalized it will be represented as the following in the database (`custom_representation column`):
+
+```json
+{
+  "net.corda.v5.ledger.utxo.ContractState" : {
+    "stateRef": "<TransactionID>:<StateIndex>"
+  },
+  "com.r3.corda.demo.contract.TestState" : {
+    "testField": ""
+  } 
+}
+```
+
+{{< note >}}
+The `net.corda.v5.ledger.utxo.ContractState` field is a part of the JSON representation for all state types.
+{{< /note >}}
+{{< note >}}
+The implementation of the JSON factory needs to be defined in the same CPK as the state
+it is working on so the platform can get hold of it when persisting a state to the database.
```

Comment:
```suggestion
that it is working on, so that the platform can get hold of it when persisting a state to the database.
```

Reply:
Ask SME does a replacement work for this section: "....platform can get hold of it...", such as "....platform can locate/find/use it...". Thanks.